### PR TITLE
Added Command Line interface 

### PR DESCRIPTION
**Pull request by mfussi66**
The pull request adds a small application that can be used on the terminal to set and get the ultrapython cameras' values.

### CODE REVIEW

File: src/cli/UltraPythonCli.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ */
+
+#include "UltraPythonCli.h"
+
+UltraPythonCli::UltraPythonCli(yarp::dev::IFrameGrabberControls* grabber){};
+
+UltraPythonCli::~UltraPythonCli() {}
+
+bool UltraPythonCli::ParseArgs(int argc, char* argv[],
+                               std::map<std::string, std::string>& args_map) {
```
Comment (triccyx) [inline src/cli/UltraPythonCli.cpp:26]: Please use camelcase name convention

File: src/cli/UltraPythonCli.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    args_map.insert({argv[3], argv[4]});
+  }
+
+  return true;
+}
+
+bool UltraPythonCli::InitYarpCommunication(const std::string& remotePort) {
+  if (!yarp::os::NetworkBase::checkNetwork(2)) {
+    std::cout
+        << "Yarp yarpserver not found.\nPlease activate yarpserver and retry."
```
Comment (triccyx) [inline src/cli/UltraPythonCli.cpp:61]: Please use std::end line instead of \n

File: src/cli/UltraPythonCli.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  /**
+   * @brief Used to parse command line arguments
+   *
+   * @param argc
+   * @param argv
+   * @param args_map map of arguments: can be --help, --remote, --set, --get.
+   * @return true if arguments could be parsed correctly, false otherwise.
+   */
+  bool ParseArgs(int argc, char* argv[], std::map<std::string, std::string>& args_map_);
```
Comment (triccyx) [inline src/cli/UltraPythonCli.h:58]: Please use camelcase

File: src/cli/main.cpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ */
+
+#include "UltraPythonCli.h"
+
+/**
+ * @brief Executes the main loop, and parses command line arguments
+ * @returns -1 on error, 0 if successful
+ */
+int main(int argc, char* argv[]) {
+  int control_code = 0;
```
Comment (triccyx) [inline src/cli/main.cpp:26]: please use camelcase

File: src/cli/main.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+  if (args_map.find("--set") != args_map.end()) {
+    std::vector<std::string> set_args = client.splitString(args_map["--set"], "=");
+
+    // Assume control code as first string
+    try {
+      control_code = std::stoi(set_args[0]);
+    } catch (const std::exception& e) {
+      std::cout << e.what()
+                << "\nControl codes can be expressed only in integer values."
```
Comment (triccyx) [inline src/cli/main.cpp:49]: use std::endl instead \n

File: src/cli/main.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  if (args_map.find("--get") != args_map.end()) {
+    try {
+      control_code = std::stoi(args_map["--get"]);
+    } catch (const std::exception& e) {
+      std::cout << e.what()
+                << "\nControl codes can be expressed only in integer values."
+                << std::endl;
+      return -1;
+    }
+    bool result = grabber->getFeature(control_code, &value);
```
Comment (triccyx) [inline src/cli/main.cpp:82]: Use camel case

File: src/devices/ultrapython/unittest/testCli.cpp [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+TEST(UltraPyCli, split_string_ok) {
+    // use ; as delimiter
+    std::string test1 = "a;b";
+    std::string test2 = "a;b;c";
+    std::string test3 = "a b";
+    std::string test4 = "a;b c";
+    std::string test5 = "a\nb;c";
+
+    IFrameGrabberControlsMock *grabber_mock = new IFrameGrabberControlsMock();
```
Comment (triccyx) [inline src/devices/ultrapython/unittest/testCli.cpp:65]: Please use camel case

**Comment by pattacini** [comment]
Hi @mfussi66, don't forget to assign yourself to the PR and request a review from the most appropriate developer.

**Comment by pattacini** [comment]
Nice that you squashed all your dev into one commit 🚀

File: src/cli/main.cpp (hunk 6 above)
Comment (mfussi66) [reply src/cli/main.cpp]: ✔️

File: src/devices/ultrapython/unittest/testCli.cpp (hunk 7 above)
Comment (mfussi66) [reply src/devices/ultrapython/unittest/testCli.cpp]: ✔️

**Pull request merged**